palace/drivers/electrostaticsolver.cpp

```diff
@@ -136,33 +134,29 @@ void ElectrostaticSolver::PostprocessTerminals(
   mfem::DenseMatrix C(V.size()), Cm(V.size());
   for (int i = 0; i < C.Height(); i++)
   {
-    // Diagonal: C_ii = 2 U_e(V_i) / V_i².
-    C(i, i) = Cm(i, i) = 2.0 * E_elec[i];
-  }
+    // Diagonal: Cᵢᵢ = 2 Uₑ(Vᵢ) / Vᵢ² = (Vᵢᵀ K Vᵢ) / Vᵢ²
+    auto &V_gf = postop.GetVGridFunction().Real();
+    auto &D_gf = postop.GetDomainPostOp().D;
+    V_gf.SetFromTrueDofs(V[i]);
+    postop.GetDomainPostOp().M_elec->Mult(V_gf, D_gf);
+    C(i, i) = Cm(i, i) = linalg::Dot<Vector>(postop.GetComm(), V_gf, D_gf);
```

Comment:
For posterity, no normalization required because `V_i = 1` for all `i`.

Reply:
[8c7ea22](https://github.com/awslabs/palace/pull/237/commits/8c7ea2254f5e7bd961c4b6c4622d4e892fb70f9b)